src/js/components/modals/AppModalComponent.jsx

```diff
@@ -165,11 +175,15 @@ var AppModalComponent = React.createClass({
 
     // Ports should always be an Array.
     if ("ports" in modelAttrs) {
-      var portStrings = modelAttrs.ports.split(",");
-      modelAttrs.ports = portStrings.map(function (p) {
-        var port = parseInt(p, 10);
-        return isNaN(port) ? p : port;
-      });
+      if (modelAttrs.ports === "") {
```

Comment:
Can't we avoid these nested `if`-statement with this more sane default? Maybe something like this:
```
if (modelAttrs.ports == null || modelAttrs.ports == "") {
  modelAttrs.ports = [];
}

let portStrings = modelAttrs.ports.split(",");
modelAttrs.ports = portStrings.map(function (p) {
  var port = parseInt(p, 10);
  return isNaN(port) ? p : port;
});
```

Same goes for the above